### PR TITLE
x265: update to 3.6; revbump dependents

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

1. Update x265, finally.
2. Revbump dependents.
3. Address a problem with FFMpeg which is pending for 5 years.

I do not revbump one completely broken port, it is pointless, and dropped `libheif` due to https://github.com/macports/macports-ports/pull/25265
I also do not include FFMpeg7 and -devel, since we rather merge https://github.com/macports/macports-ports/pull/25173 after this PR and avoid another rebuild.

###### Type(s)

- [x] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

File: multimedia/VLC2/Portfile [hunk 1]
```diff
@@ -175,6 +175,10 @@ if {(${subport} eq ${name}) || (${subport} eq "lib${name}")} {
     } elseif {[string match *gcc* ${configure.compiler}]} {
         patchfiles-append \
                         0001-Fixes-for-modern-GCC.patch
+        # codec/avcodec/encoder.c:883:41: error: passing argument 2 of 'vlc_av_get_options_ng'
+        # from incompatible pointer type [-Wincompatible-pointer-types]
+        configure.cflags-append \
+                        -Wno-incompatible-pointer-types
```
Comment (pmetzger) [inline multimedia/VLC2/Portfile:181]: I really hate removing compiler warnings. I think it's important to know why something fails and not just decide to ignore the compiler.

Comment (barracuda156) [reply multimedia/VLC2/Portfile:181]: This is an old version of VLC, it is not supported by upstream.

Comment (pmetzger) [reply multimedia/VLC2/Portfile:181]: Fine, then patch the code instead of ignoring the warning?

Comment (barracuda156) [reply multimedia/VLC2/Portfile:181]: @pmetzger If you suggest a fix, I will happily add it.

Comment (pmetzger) [reply multimedia/VLC2/Portfile:181]: Read the code at that place. Is the pointer being passed safe to pass? If so, add a coercion. Is it unsafe to pass? If so, figure out what sort of patch would render the code correct.

Comment (barracuda156) [reply multimedia/VLC2/Portfile:181]: @pmetzger Thank you for insisting here, I think we actually had an error in one patch. I will try sorting that out now.

Comment (barracuda156) [reply multimedia/VLC2/Portfile]: @pmetzger It took way longer than expected, but now VLC2 builds with gcc14 without converting errors into warnings.

**Pull request merged**